rigid_transformations.py

```diff
@@ -0,0 +1,611 @@
+"""Computations on the Lie group of rigid transformations."""
+
+import logging
+import numpy as np
+
+import rotations as rotations
+
+TRANSLATIONS_IDENTITY = np.array([0, 0, 0])
+
+GROUP_IDENTITY = np.concatenate([rotations.GROUP_IDENTITY,
+                                 TRANSLATIONS_IDENTITY])
+ALGEBRA_CANONICAL_INNER_PRODUCT = np.eye(6)
+
+
+def regularize_transformation(transfo):
+    """
+    Regularize an element of the group SE(3),
+    by extracting the rotation vector r from the input [r t]
+    and using rotations.regularize_rotation_vector.
+
+    :param transfo: 6d vector, element in SE(3) represented as [r t].
+    :returns regularized_transfo: 6d vector, element in SE(3)
+    with regularized rotation.
+    """
+
+    regularized_transfo = transfo
+    rot_vec = transfo[0:3]
+    regularized_transfo[0:3] = rotations.regularize_rotation_vector(rot_vec)
+
+    return regularized_transfo
+
+
+def inverse(transfo):
+    """
+    Compute the group inverse in SE(3).
+
+    Formula:
+    TODO(nina).
+
+    :param transfo: 6d vector element in SE(3)
+    :returns inverse_transfo: 6d vector inverse of transfo
+    """
+
+    transfo = regularize_transformation(transfo)
+
+    rot_vec = transfo[0:3]
+    translation = transfo[3:6]
+
+    inverse_transfo = np.zeros(6)
+    inverse_transfo[0:3] = -rot_vec
+    rot_mat = rotations.rotation_matrix_from_rotation_vector(-rot_vec)
+    inverse_transfo[3:6] = np.dot(rot_mat, -translation)
+
+    inverse_transfo = regularize_transformation(inverse_transfo)
+    return inverse_transfo
+
+
+def compose(transfo_1, transfo_2):
+    """
+    Compose two elements of group SE(3).
+
+    Formula:
+    transfo_1 . transfo_2 = [R1 * R2, (R1 * t2) + t1]
+    where:
+    R1, R2 are rotation matrices,
+    t1, t2 are translation vectors.
+
+    :param transfo_1, transfo_2: 6d vectors elements of SE(3)
+    :returns prod_transfo: composition of transfo_1 and transfo_2
+    """
+    rot_mat_1 = rotations.rotation_matrix_from_rotation_vector(transfo_1[0:3])
+    rot_mat_1 = rotations.closest_rotation_matrix(rot_mat_1)
+
+    rot_mat_2 = rotations.rotation_matrix_from_rotation_vector(transfo_1[0:3])
+    rot_mat_2 = rotations.closest_rotation_matrix(rot_mat_2)
+
+    translation_1 = transfo_1[3:6]
+    translation_2 = transfo_2[3:6]
+
+    prod_transfo = np.zeros(6)
+    prod_rot_mat = np.dot(rot_mat_1, rot_mat_2)
+
+    prod_transfo[0:3] = rotations.rotation_vector_from_rotation_matrix(
+            prod_rot_mat)
+    prod_transfo[3:6] = np.dot(rot_mat_1, translation_2) + translation_1
+
+    prod_transfo = regularize_transformation(prod_transfo)
+    return prod_transfo
+
+
+def jacobian_translation(transfo, left_or_right='left'):
```

Comment:
what about direction instead of left_or_right?

Reply:
Left or Right is the actual mathematical term: it comes from the left- or right- translations associated to the group composition resp. by the left- or the right-.

Reply:
cool thanks for the clarification